src/presets/vercel.ts

```diff
@@ -11,38 +11,23 @@ export const vercel = defineNitroPreset({
   entry: '#internal/nitro/entries/vercel',
   output: {
     dir: '{{ rootDir }}/.vercel/output',
-    serverDir: '{{ output.dir }}/functions/index.func',
+    serverDir: '{{ output.dir }}/functions/__nuxt.func',
     publicDir: '{{ output.dir }}/static'
   },
   commands: {
     deploy: '',
     preview: ''
   },
   hooks: {
+    'prerender:generate' (route, nitro: any) {
+      nitro._routes = nitro._routes || new Set<string>()
```

Comment:
We could move it to prerender.ts (but call it `_routeFiles` or something like this)